diff --git a/code/game/g_main.c b/code/game/g_main.c
index 4bf2fd5..e6199f5 100644
--- a/code/game/g_main.c
+++ b/code/game/g_main.c
@@ -1978,6 +1978,12 @@ void CheckExitRules( void ) {
 		}
 	}
 
+	if ( g_fraglimit.integer < 0 ) {
+		trap_SendServerCommand( -1, "print \"Fraglimit is negative\n\"" );
+		trap_Cvar_Set( "fraglimit", "0" );
+		trap_Cvar_Update( &g_fraglimit );
+	}
+
 	if ( g_gametype.integer < GT_RTP && g_gametype.integer != GT_DUEL &&
 		g_fraglimit.integer ) {
 		if ( level.teamScores[TEAM_RED] >= g_fraglimit.integer ) {
